components/brave_rewards/browser/rewards_service_browsertest.cc

```diff
@@ -185,12 +101,29 @@ class BraveRewardsBrowserTest : public InProcessBrowserTest {
  public:
   void SetUpOnMainThread() override {
     InProcessBrowserTest::SetUpOnMainThread();
+    brave::RegisterPathProvider();
+    ReadTestData();
   }
 
   void TearDown() override {
     InProcessBrowserTest::TearDown();
   }
 
+  void ReadTestData() {
+    base::FilePath path;
+    ASSERT_TRUE(base::PathService::Get(brave::DIR_TEST_DATA, &path));
+    path = path.AppendASCII("rewards-data");
+    ASSERT_TRUE(base::PathExists(path));
+    base::ReadFileToString(path.AppendASCII("register_persona_resp.json"),
```

Comment:
Does this work on debug build w/o `base::AssertBlockingAllowed()`?

Reply:
not sure cc @bbondy

Reply:
@jasonrsadler I found the reason why io is allowed here.
The reason is `SetUpOnMainThread()` is called before io is disabled on main thread.
During the startup, IO is disabled on main thread by `base::ThreadRestrictions::SetIOAllowed(false);` in the below method.

SetUpOnMainThread() was called by `parts_->PreMainMessageLoopRun();` before `SetIOAllowed(false)`.
Because of this reason, file operation api can be used in `SetUpOnMainThread()`.

```
int BrowserMainLoop::PreMainMessageLoopRun() {
#if defined(OS_ANDROID)
  // Let screen instance be overridable by parts.
  ui::SetScreenAndroid();
#endif

  if (parts_) {
    TRACE_EVENT0("startup",
        "BrowserMainLoop::CreateThreads:PreMainMessageLoopRun");

    parts_->PreMainMessageLoopRun();
  }

  // If the UI thread blocks, the whole UI is unresponsive.
  // Do not allow disk IO from the UI thread.
  base::ThreadRestrictions::SetIOAllowed(false);
  base::ThreadRestrictions::DisallowWaiting();
  return result_code_;
}
```